src/mixins/itext_behavior.mixin.js

```diff
@@ -430,8 +430,8 @@
       this.hiddenTextarea.selectionEnd = this.selectionEnd;
       if (this.selectionStart === this.selectionEnd) {
         var p = this._calcTextareaPosition();
```

Comment:
modify _calcTextareaPosition to take in account the viewporttransform.

before returning the point, use fabric.util.transformPoint(p, this.getViewportTransform());

so then we can use this modified function to take in account also the other case.